### DISCUSSION TITLE
cellular_modem sample for Quectel EG800Q

### DISCUSSION THREAD

**Discussion by MulattoKid** [zephyrproject-rtos/zephyr#67417] [Q&A]
Hi,

I'm working on getting the `cellular_modem` sample working on an MIMXRT1040-EVK with a Quectel EG800Q modem. The EG800Q and BG95 should be quite similar, so I tried building the `modem_cellular` driver for that modem. I ran into a problem where the dial-up (`ATD*99***1#`) would only work if I had a breakpoint between each AT command being sent. When just running normally I get a `NO CARRIER` response.

After some testing, I've found it to be due to `AT+CFUN=4` in `init_chat_script` and `AT+CFUN=1` in the `dial_chat_script`. If I remove `AT+CFUN=4` the dial-up is able to connect.

My question is why the `AT+CFUN=4` command is needed in the first place?
cc @bjarki-trackunit

**Comment by Mauricio-Alencar** [comment]
Hi @MulattoKid ,

You dont need to exeute AT+CFUN=4 first. AT+CFUN=4 put the modem in airplane mode and disable LTE antena circuits.

If you dont need to put in airplane mode, remove the AT+CFUN=4 in your logic and enter direclty in AT+CFUN=1.

**Comment by bjarki-andreasen** [comment]
The reason AT+CFUN=4 is sent is to "reset" the modem, closing any ongoing dial out andclosing pdp contexts. This is not always neccessary and sometimes even breaking, this is why there are custom chat scripts for each modem, so they can be tailored correctly :)

**Comment by MulattoKid** [comment]
Thanks for the clarification :)
